Fix _cat operator

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34591 Fix _cat operator**

Differential Revision: [D20388000](https://our.internmc.facebook.com/intern/diff/D20388000)

VitalyFedyunin committed:
Fix _cat operator

[ghstack-poisoned]

albanD reviewed (approved):
Interesting.
What is the impact of the missing `native::` here? Only that it goes through dispatch again?

VitalyFedyunin commented:
@pytorchbot retest this please

VitalyFedyunin commented:
@pytorchbot retest this please

VitalyFedyunin committed:
Update on "Fix _cat operator"


Differential Revision: [D20388000](https://our.internmc.facebook.com/intern/diff/D20388000)

[ghstack-poisoned]
